packages/x-data-grid/src/components/toolbar/GridToolbarColumnsButton.tsx

```diff
@@ -15,7 +14,10 @@ interface GridToolbarColumnsButtonProps {
    * The props used for each slot inside.
    * @default {}
    */
-  slotProps?: { button?: Partial<ButtonProps>; tooltip?: Partial<TooltipProps> };
+  slotProps?: {
+    button?: Partial<GridSlotProps['baseButton']>;
+    tooltip?: Partial<GridSlotProps['baseTooltip']>;
+  };
```

Comment:
This has the same issue mentioned in other places, those components are now base components and not material components, so if a user tries to pass material-ui props, they'll be disappointed.